### PR TITLE
Add Steady State Utils

### PR DESCRIPTION
**Pull request by justusschock**
This adds steady state utils and removes windows tests suite at the same time as windows tests do not make sense for a linux-based cluster system

### CODE REVIEW

File: lit_llms/callbacks/steady_state_detection.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        target_loss: float,
+        batch_size: Optional[int] = None,
+        num_params: Optional[int] = None,
+        rtol: float = 0.015,
+        atol: Optional[float] = None,
+        steady_state_det_mode: str = "iter_speed",
+        average: Optional[float] = None,
+        moving_average_window: int = 10,
+        stop_on_steady_state: bool = True,
+        steady_state_steps_before_stop: int = 10,
```
Comment (awaelchli) [inline lit_llms/callbacks/steady_state_detection.py:26]: @justusschock Should there be something like a max_steps?
What if steady state can't be achieved and I don't want to run for too long?

**Review by awaelchli — COMMENTED** [review]
🎉 💪

File: lit_llms/callbacks/steady_state_detection.py (hunk 1 above)
Comment (awaelchli) [reply lit_llms/callbacks/steady_state_detection.py:26]: Maybe that's not a realistic enough scenario

Comment (justusschock) [reply lit_llms/callbacks/steady_state_detection.py:26]: I think there shouldn't. This should run in parallel with a max-steps setting of the trainer or an early-stopping callback.

**Review by awaelchli — APPROVED** [review]

**Pull request merged**